### DISCUSSION TITLE
Public Constructor for IfcRightCircularCone

### DISCUSSION THREAD

**Discussion by jimbosquash** [GeometryGym/GeometryGymIFC#66] [Q&A]
Hi there!

your library is fantastic and I have been appreciative of the extensive work you have done!

I am now trying to Create Cones to use in a boolean operation and I have found that there is no public constructor whilst there is a public constructor for other primitives (Spheres, blocks). Are there any plans to implement a constructor?

 I would be very happy to help implement this if I am able to assist. I do plan to keep working with your library so if I can help and contribute I will.

Thank you in advance,
James

**Comment by jmirtsch** [comment]
Hi James,

Thanks for posting (and the positive feedback).  The original work was internal, I only open sourced the work at a later point, so as you've identified some constructors are not public.  If you notice anything like this, please feel free to adjust the code and make a pull request.  External contributions are appreciated, else post an issue listing suggestions and I'll review and implement.

Thanks,

Jon
